NitroxPatcher/Patches/Persistent/uGUI_OptionsPanel_AddTabs_Patch.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Harmony;
+using NitroxClient.MonoBehaviours.Gui.Settings;
+using System;
+using System.Reflection;
+using UnityEngine.Events;
+
+namespace NitroxPatcher.Patches.Persistent
+{
+    public class uGUI_OptionsPanel_AddTabs_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(uGUI_OptionsPanel);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("AddTabs", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        public static void Postfix(uGUI_OptionsPanel __instance)
+        {
+            SettingsOptions settingsOptions = new SettingsOptions();
+
+            int tabIndex = __instance.AddTab("Multiplayer");
+            settingsOptions.AddInputFieldOption(tabIndex, "Name", "Enter Name", SettingsManager.GetName(), __instance, new UnityAction<string>(SettingsManager.SetName));
+            settingsOptions.AddKeybindOption(tabIndex, "Chat", SettingsManager.GetKey_Chat(), __instance, new UnityAction<string>(SettingsManager.SetKey_Chat));
+            __instance.AddHeading(tabIndex, "Player color");
+            __instance.AddSliderOption(tabIndex, "Red", SettingsManager.GetColorR(), new UnityAction<float>(SettingsManager.SetColorR));
+            __instance.AddSliderOption(tabIndex, "Green", SettingsManager.GetColorG(), new UnityAction<float>(SettingsManager.SetColorG));
+            __instance.AddSliderOption(tabIndex, "Blue", SettingsManager.GetColorB(), new UnityAction<float>(SettingsManager.SetColorB));
```

Comment:
We should tuck some of this logic into a class to keep the patchers slime

Reply:
You could for instance store all this logic in a monobehaviour that encapsulates the multiplayer menu. This also allows to store often-used variables such as uGUI_Control and tabIndex in a field.